### DISCUSSION TITLE
How to implement domain specific features - like Media Session control

### DISCUSSION THREAD

**Discussion by yschimke** [mobile-dev-inc/Maestro#911] [Ideas]
I'm testing media apps, as well as controlling the UI, I'd like to control media playback through the Android Media Session APIs.  

https://developer.android.com/guide/topics/media/media3/getting-started/mediasession

Is there a way to extend the vocabulary of the Maestro scripting language for new features like this?

Or does it require this as a supported feature for Maestro, including across platforms.

**Comment by yschimke** [comment]
Would contributions be possible? or is this too complex?

**Comment by axelniklasson** [comment]
Hi there @yschimke! Maestro is a blackbox UI testing framework so we try to stay away from interacting directly with native APIs. I'd recommend interacting with your app like your users would in your tests.
